### PR TITLE
Bug 1825549 Disallow messages with blank text fields

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [Bug 1825549](https://bugzilla.mozilla.org/show_bug.cgi?id=1825549).

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1825549

### CODE REVIEW

File: android-components/components/service/nimbus/src/main/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorage.kt [hunk 1]
```diff
@@ -54,10 +54,17 @@ class NimbusMessagingStorage(
     /**
      * Returns the [Message] for the given [key] or returns null if none found.
      */
+    suspend fun getMessage(key: String): Message? = createMessage(key)
```
Comment (t-p-white) [inline android-components/components/service/nimbus/src/main/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorage.kt:57]: drive by - what benefit does having the extra `getMessage` func have over just using a public `createMessage` ?

Comment (jhugman) [reply android-components/components/service/nimbus/src/main/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorage.kt]: Oh good spot. I've changed this to re-use the accept a `featureValue`.

File: android-components/components/service/nimbus/src/test/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorageTest.kt [hunk 2]
```diff
@@ -57,6 +60,7 @@ class NimbusMessagingStorageTest {
     fun setup() {
         MockitoAnnotations.openMocks(this)
         malformedWasReported = false
+        malformedMessageIds.clear()
```
Comment (Amejia481) [inline android-components/components/service/nimbus/src/test/java/mozilla/components/service/nimbus/messaging/NimbusMessagingStorageTest.kt:63]: Nit: Maybe we could have it inside  a function with [@After](https://junit.org/junit4/javadoc/4.12/org/junit/After.html) maybe `teardown` or `after` :)

**Review by Amejia481 — APPROVED** [review]
LGTM!

**Pull request merged**